src/test/java/net/datafaker/idnumbers/HungarianIdNumberTest.java

```diff
@@ -0,0 +1,34 @@
+package net.datafaker.idnumbers;
+
+import org.junit.jupiter.api.Test;
+
+import static net.datafaker.providers.base.PersonIdNumber.Gender.FEMALE;
+import static net.datafaker.providers.base.PersonIdNumber.Gender.MALE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class HungarianIdNumberTest {
+
+    @Test
+    void firstDigit() {
+        assertThat(HungarianIdNumber.firstDigit(1899, MALE)).isEqualTo(3);
+        assertThat(HungarianIdNumber.firstDigit(1900, MALE)).isEqualTo(1);
+        assertThat(HungarianIdNumber.firstDigit(1901, MALE)).isEqualTo(1);
+        assertThat(HungarianIdNumber.firstDigit(1899, FEMALE)).isEqualTo(4);
+        assertThat(HungarianIdNumber.firstDigit(1900, FEMALE)).isEqualTo(2);
+        assertThat(HungarianIdNumber.firstDigit(1901, FEMALE)).isEqualTo(2);
+        assertThat(HungarianIdNumber.firstDigit(1999, MALE)).isEqualTo(1);
+        assertThat(HungarianIdNumber.firstDigit(2000, MALE)).isEqualTo(3);
+        assertThat(HungarianIdNumber.firstDigit(1999, FEMALE)).isEqualTo(2);
+        assertThat(HungarianIdNumber.firstDigit(2000, FEMALE)).isEqualTo(4);
+    }
+
+    @Test
+    void checkDigit() {
+        assertThat(HungarianIdNumber.getCheckDigit("807159215")).isEqualTo(3);
+        assertThat(HungarianIdNumber.getCheckDigit("100000000")).isEqualTo(1);
+        assertThat(HungarianIdNumber.getCheckDigit("010000000")).isEqualTo(2);
+        assertThat(HungarianIdNumber.getCheckDigit("001000000")).isEqualTo(3);
+        assertThat(HungarianIdNumber.getCheckDigit("000000001")).isEqualTo(9);
+        assertThat(HungarianIdNumber.getCheckDigit("170010103")).isEqualTo(10);
```

Comment:
If it's 10 wouldn't that make the whole value 11 char?

Reply:
Yes, that's why we need this loop in `HungarianIdNumber.generateValid`:

```java
        do {
            ...
            checkDigit = getCheckDigit(...);
        } while (checkDigit == 10);
```

Reply:
Thanks, sorry I have a bad habit of overlooking conditions in do whiles (that's on me).